runtime/src/main/scala/akka/grpc/internal/AkkaHttpClientUtils.scala

```diff
@@ -0,0 +1,299 @@
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+object AkkaHttpClientUtils {
+
+  /**
+   * INTERNAL API
+   */
+  @InternalApi
+  def createChannel(settings: GrpcClientSettings, log: LoggingAdapter)(
+      implicit sys: ClassicActorSystemProvider): InternalChannel = {
+    implicit val ec = sys.classicSystem.dispatcher
+
+    log.debug("Creating gRPC client channel")
+
+    // https://github.com/grpc/grpc/blob/master/doc/compression.md
+    // since a client can't assume what algorithms a server supports, we
+    // must default to no compression.
+    // Configuring a different default could be a future feature.
+    // Configuring compression per call could be a future power API feature.
+    implicit val writer = GrpcProtocolNative.newWriter(Identity)
+
+    val target = {
+      // TODO FIXME discovery, loadbalancing etc
+      // https://github.com/akka/akka-grpc/issues/1196
+      // https://github.com/akka/akka-grpc/issues/1197
+      val resolved = settings.serviceDiscovery.lookup(settings.serviceName, 10.seconds)
+      Await.result(resolved, 10.seconds).addresses.head
+    }
+
+    val clientConnectionSettings = settings.overrideAuthority match {
+      case None => ClientConnectionSettings(sys)
+      case Some(authority) =>
+        ClientConnectionSettings(sys).withTransport(ClientTransport.withCustomResolver((host, port) => {
+          assert(host == authority)
+          assert(port == target.port.getOrElse(settings.defaultPort))
+          target.address match {
+            case Some(address) =>
+              Future.successful(new InetSocketAddress(address, target.port.getOrElse(settings.defaultPort)))
+            case None =>
+              Future.successful(new InetSocketAddress(target.host, target.port.getOrElse(settings.defaultPort)))
+          }
+
+        }))
+    }
+
+    val builder = Http()
+      .connectionTo(settings.overrideAuthority.getOrElse(target.host))
+      .toPort(target.port.getOrElse(settings.defaultPort))
+      .withClientConnectionSettings(clientConnectionSettings)
+
+    val http2client =
+      if (settings.useTls) {
+        val connectionContext =
+          ConnectionContext.httpsClient {
+            settings.trustManager match {
+              case None => SSLContext.getDefault
+              case Some(trustManager) =>
+                val sslContext: SSLContext = SSLContext.getInstance("TLS")
+                sslContext.init(Array[KeyManager](), Array[TrustManager](trustManager), new SecureRandom)
+                sslContext
+            }
+          }
+
+        builder.withCustomHttpsConnectionContext(connectionContext).http2()
+      } else {
+        builder.http2WithPriorKnowledge()
+      }
+
+    val (queue, doneFuture) =
+      Source
+        .queue[HttpRequest](4242, OverflowStrategy.fail)
+        .via(http2client)
+        .toMat(Sink.foreach { res =>
+          res.attribute(ResponsePromise.Key).get.promise.trySuccess(res)
+        })(Keep.both)
+        .run()
```

Comment:
So, right now we create a single client and when that dies, the channel is dead?